src/dataLayer/mongo/user.js

```diff
@@ -1,46 +1,67 @@
 import validator from 'validator';
 import UserModel from '../model/user.js';
 import debug from 'debug';
+import uuid from 'uuid/v4';
+import { isEmpty, isString } from 'lodash';
 
 import { asyncErrorHandler } from '../../utils';
+import { verifyWebToken, namespace } from '../../auth';
 
 const log = debug('fcc:dataLayer:mongo:user');
 
 function doesExist(Model, options) {
   return Model.find(options).exec();
 }
 
-export function getUsers(_, args = {}) {
-  return new Promise((resolve, reject) => {
-    UserModel.find(args)
-      .then(users => {
-        let userMap = [];
-        users.forEach((user, index) => {
-          userMap[index] = user;
-        });
-        let res = {};
-        res.users = userMap;
-        resolve(res.users);
-      })
-      .catch(err => {
-        reject(err);
-      });
+export function getUser(root, { email }) {
+  return new Promise(async function getUserPromise(resolve, reject) {
+    if (!isString(email) || !validator.isEmail(email)) {
+      reject(
+        new TypeError(`Expected a valid email, got ${JSON.stringify(email)}`)
+      );
+      return null;
+    }
+    log(`finding user for ${email}`);
+    const found = await UserModel.findOne({ email }).exec();
+    log(`found? ${!!found}`);
+    if (isEmpty(found)) {
+      reject(new Error(`No user found for ${email}`));
+      return null;
+    }
+    return resolve(found);
   });
 }
 
-export async function createUser(email) {
+export async function createUser(root, vars, ctx) {
+  const { decoded } = verifyWebToken(ctx);
+  const { email, name } = decoded;
   const isEmail = validator.isEmail(email);
   if (!isEmail) {
     throw new Error('You must provide a vaild email');
   }
-  const entities = await asyncErrorHandler(doesExist(UserModel, { email }));
-  const exists = !!entities.length;
-  if (exists) {
-    throw new Error(`An account with the email ${email} already exists`);
+  const newUser = { name, email };
+  const accountLinkId = decoded[namespace + 'accountLinkId'];
+  if (accountLinkId) {
+    newUser.accountLinkId = accountLinkId;
+  } else {
+    newUser.accountLinkId = uuid();
+    // TODO: appy accountLinkId to app_metadata (auth0)
```

Comment:
I will fix this typo by adding the code required to talk to auth0

Reply:
Haha! That's a good one.!